src/tribol/interface/mfem_tribol.cpp

```diff
@@ -119,6 +118,133 @@ void setMfemLORFactor( integer cs_id,
+   );
+   setPenaltyOptions(cs_id, KINEMATIC, KINEMATIC_ELEMENT);
+   coupling_scheme->getMfemMeshData()->ClearAllPenaltyData();
+   coupling_scheme->getMfemMeshData()->ComputeElementThicknesses();
+   coupling_scheme->getMfemMeshData()->SetMaterialModulus(modulus_coefficient);
+}
+
+void setMfemRateConstantPenalty( integer cs_id, 
+                                 real mesh1_penalty, 
+                                 real mesh2_penalty )
+{
+   auto coupling_scheme = CouplingSchemeManager::getInstance().getCoupling(cs_id);
+   SLIC_ERROR_ROOT_IF(
+      !coupling_scheme->hasMfemData(),
+      "Coupling scheme does not contain MFEM data. "
+      "Create the coupling scheme using registerMfemCouplingScheme() to set the penalty."
+   );
+   auto penalty_opts = coupling_scheme->getEnforcementOptions().penalty_options;
+   SLIC_ERROR_ROOT_IF(
+      !penalty_opts.kinematic_calc_set,
+      "No kinematic enforcement method set. Call setMfemKinematicConstantPenalty() or "
+      "setMfemKinematicElementPenalty() first."
+   );
+   setPenaltyOptions(
+      cs_id, penalty_opts.constraint_type, penalty_opts.kinematic_calculation, RATE_CONSTANT);
+   coupling_scheme->getMfemMeshData()->ClearRatePenaltyData();
+   coupling_scheme->getMfemMeshData()->SetMesh1RatePenalty(mesh1_penalty);
+   coupling_scheme->getMfemMeshData()->SetMesh2RatePenalty(mesh2_penalty);
+}
+
+void setMfemRatePercentPenalty( integer cs_id, 
+                                 real mesh1_scale, 
+                                 real mesh2_scale )
+{
+   auto coupling_scheme = CouplingSchemeManager::getInstance().getCoupling(cs_id);
+   SLIC_ERROR_ROOT_IF(
+      !coupling_scheme->hasMfemData(),
+      "Coupling scheme does not contain MFEM data. "
+      "Create the coupling scheme using registerMfemCouplingScheme() to set the penalty."
+   );
+   auto penalty_opts = coupling_scheme->getEnforcementOptions().penalty_options;
+   SLIC_ERROR_ROOT_IF(
+      !penalty_opts.kinematic_calc_set,
+      "No kinematic enforcement method set. Call setMfemKinematicConstantPenalty() or "
+      "setMfemKinematicElementPenalty() first."
+   );
+   setPenaltyOptions(
+      cs_id, penalty_opts.constraint_type, penalty_opts.kinematic_calculation, RATE_PERCENT);
+   coupling_scheme->getMfemMeshData()->ClearRatePenaltyData();
+   coupling_scheme->getMfemMeshData()->SetMesh1RateScale(mesh1_scale);
+   coupling_scheme->getMfemMeshData()->SetMesh2RateScale(mesh2_scale);
+}
+
+void setMfemPenaltyScale( integer cs_id, real mesh1_scale, real mesh2_scale )
+{
+   auto coupling_scheme = CouplingSchemeManager::getInstance().getCoupling(cs_id);
+   SLIC_ERROR_ROOT_IF(
+      !coupling_scheme->hasMfemData(),
+      "Coupling scheme does not contain MFEM data. "
+      "Create the coupling scheme using registerMfemCouplingScheme() to set the penalty."
+   );
+   auto penalty_opts = coupling_scheme->getEnforcementOptions().penalty_options;
+   SLIC_ERROR_ROOT_IF(
+      !penalty_opts.kinematic_calc_set,
```

Comment:
Double check header file for a \pre or \note explaining that the kinematic penalty has to be set first.